src/m365/spo/commands/site/site-admin-add.spec.ts

```diff
@@ -0,0 +1,693 @@
+    sinon.stub(session, 'getId').returns('');
+    sinon.stub(spo, 'getRequestDigest').resolves({
+      FormDigestValue: 'abc',
+      FormDigestTimeoutSeconds: 1800,
+      FormDigestExpiresAt: new Date(),
+      WebFullUrl: 'https://contoso.sharepoint.com'
+    });
+    auth.connection.active = true;
+    auth.connection.spoUrl = 'https://contoso.sharepoint.com';
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.post,
+      request.patch
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+    auth.connection.spoUrl = undefined;
+  });
+
+  it('fails validation if siteUrl is not a valid SharePoint URL', async () => {
+    const actual = await command.validate({ options: { siteUrl: 'foo', userId: adminToAddId } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if userId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { siteUrl: siteUrl, userId: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if userName is not a valid UPN', async () => {
+    const actual = await command.validate({ options: { siteUrl: siteUrl, userName: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if groupId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { siteUrl: siteUrl, groupId: 'invalid' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when the siteUrl is a valid SharePoint URL and userId is a valid GUID', async () => {
+    const actual = await command.validate({ options: { siteUrl: siteUrl, userId: adminToAddId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the siteUrl is a valid SharePoint URL and userName is a valid UPN', async () => {
+    const actual = await command.validate({ options: { siteUrl: siteUrl, userName: adminToAddUPN } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('adds a user to site collection admins by userId as admin', async () => {
+    let result = false;
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/sites/root?$select=webUrl`) {
+        return { res: { webUrl: rootUrl } };
+      }
+
+      if (opts.url === `https://graph.microsoft.com/v1.0/users/${adminToAddId}`) {
+        return { userPrincipalName: adminToAddUPN };
+      }
```

Comment:
You will have to replace this when changing to entraUser.

Reply:
done